### ISSUE TITLE
RFC 5 coordinate transformations

### ISSUE THREAD

**Issue by d-v-b** [BioImageTools/ome-zarr-models-py#54]
It would be great to start implementing some of the data structures defined in the coordinate transformations proposal, assuming there won't be too much churn in those data structures. I _think_ the models we create should be based on the proposed spec changes in [this PR](https://github.com/ome/ngff/pull/138), @bogovicj please correct me if I'm looking in the wrong place.

in terms of scope, we should only aim to model the structure of the transformations, not their behavior.

**Comment by bogovicj** [comment]
Yup, that's the place to look.

Looking forward to watching this repo!

**Comment by LucaMarconato** [comment]
I'd like to start working on this together with @melonora and add a `src/ome_zarr_models/_rfc5` location in the code.

Regarding the development, instead of making a single large PR, I'd like to split the work into smaller PRs (for instance starting with translation + scale and then generalizing).

Which of these two approaches do you prefer?
1) fork main, open small PRs against **my** fork of main (so always working on my fork), and open a single final PR against this repo once everything is ready;
2) or to have a branch in this repo, open PRs against the branch (in this repo). 

If 2) is the preferred way, shall I also open a draft PR against main (please mind that it will stay open for a while), or shall I open such PR only toward the end of the full implementation?

Thanks!

**Comment by dstansby** [comment]
Exciting!

So I think as an end goal we want to end up with one branch that implements RFC5 - I've created an `rfc5` branch for this. The process then would be:

- Develop as the RFC is developed on the `rfc5` branch
- Once the RFC is accepted, and becomes part of the SPEC, we merge it into main.

Because RFCs should be merged into the next major release (presumably version 0.6...?), we also need to create a `ome_zarr_models.v06` submodule too for you to work off. This will start by just inheriting from `v05` to start with, then any modifications will require copy/pasting the classes being modified to the `v06` namespace, then modifying them (e.g., see https://github.com/ome-zarr-models/ome-zarr-models-py/pull/124 for a v04 -> v05 transition).

A slight blocker here is we haven't implemented `v05` yet, but it is work in progress! I will ping you when it's done, I think at this point it's easier for me to just finish that off, and also create an empty `v06`, in the next couple of weeks.

In terms of approach (multiple vs. one PR), I guess it depends how complex the coordinate transforms RFC is? I think I would lean towards doing multiple PRs if the RFC can easily be broken down? If there are isolated parts of the RFC you can start creating models for straight away that are completely new and not modifying existing v05 data structures, they could be inmplemented straight away against this branch in a new `ome_zarr_models.v06` submodule, that you can create.

I hope that all makes sense - typing this in a slightly tired state on a train, so maybe it doesn't... but please do ask questions and bug us! We're super excited for folks to start implementing RFCs here!

**Comment by LucaMarconato** [comment]
Thanks for the explanation! Sounds great to me 😊 

Since v05 should not modify what v04 introduces for transformations (I'll double check), I think we can start by copy-pasting some of the v04 code for the transformations into an empty v06 submodule and start coding there.

**Comment by jo-mueller** [comment]
Closing this since I think they're all there :)
